### PR TITLE
re-org with fork + adding controllers, dependencies and library dir + config options

### PR DESCRIPTION
**Pull request by brynamo**
re-org with fork + adding iotDevice+plant+environment controller and dependencies. Add library dir and some config options.

### CODE REVIEW

**Comment by juanshishido** [comment]
Exciting!

Since I'm not familiar with this side of things, my first question is, how can I test that this works? Do I use a localhost? I see a `test/` directory, which is good, but I don't know how to run the tests.

File: config/config.json [hunk 1]
```diff
@@ -28,8 +26,11 @@
     
         ]
     },
-    
 
+    "databaseConfig": {
+        "host": "localhost",
+        "database": "iotTest"
```
Comment (binnie869) [inline config/config.json]: Will this be the database on mongolabs?

**Comment by binnie869** [comment]
This is cool stuff but like Juan, I would like to get some more info into what these files (environment.js and iotdevice.js) are for

File: lib/database.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            var db = mongoose.connection;
+            db.on('error', console.error.bind(console, 'db connection error:'));
+            db.once('open', function callback() {
+                console.log('db connection open');
+            });
+        }
+    };
+};
+
+module.exports = db();
```
Comment (binnie869) [inline lib/database.js]: I was thinking about adding more functions like update/add/query to perform operations on the db. These functions I assume can be used in other modules.

**Comment by brynamo** [comment]
@juanshishido + @binnie869 Lets go over the structure today (2/18/16). @binnie869 yeah those should be other modules that we can call (require) in other scripts.

**Comment by juanshishido** [comment]
Would like someone to merge this, if ready. I can do it, if I understand what things are.

Thanks!

**Comment by binnie869** [comment]
I am going to go over this with @brynamo on Sunday and then I can close this. Does that sound good?

**Comment by juanshishido** [comment]
Thanks, @binnie869! I'll see the both of you on Sunday then :+1: 

Also, only close if you don't want to accept the PR.

**Comment by brynamo** [comment]
there is very little to check in here...the main one being connection to the DB (which right now is for local not our MongoLab Instance). 
I haven't written the Mocha for it but really if someone can clone and run:

- npm install
- npm start

and see if there is no fail then this can be merged. I don't think this is a sustainable way to do things but for this second...might be productive.

**Comment by juanshishido** [comment]
Agree with @brynamo!

@binnie869 Can you please check that and post a screen shot?

**Pull request merged**